src/kernels/execution/helpers.ts

```diff
@@ -681,7 +681,7 @@ export async function updateNotebookMetadataWithSelectedKernel(
                 break;
         }
 
-        if (metadata.kernelspec?.name !== name) {
+        if (metadata.kernelspec?.name !== name || metadata.kernelspec?.display_name !== displayName) {
```

Comment:
This will introduce a regression
Changing kernel spec metadata when it shouldn’t

Reply:
But the display_name is in the kernel metadata. Changing this does actually change the contents in the .ipynb file

Reply:
Seems like a bug to me that the metadata doesn't reflect the file on disk

Reply:
Do you have any idea what the regression might be? This code never took into account the display_name AFAICT and it's rather difficult to search for bugs with 'display_name' in them because every notebook has that.

Reply:
Looking into this today.

Reply:
Yes, your change is correct.